test/apps/CMakeLists.txt

```diff
@@ -0,0 +1,112 @@
+
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin/test)
+
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}
+    $<TARGET_PROPERTY:teca_alg,INTERFACE_INCLUDE_DIRECTORIES>
+    $<TARGET_PROPERTY:teca_io,INTERFACE_INCLUDE_DIRECTORIES>
+    )
+
+teca_add_test(test_deeplabv3p_ar_detect_app
+    LIBS teca_io teca_alg
+    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/test_deeplabv3p_ar_detect_app.sh
+    ${CMAKE_BINARY_DIR}/${BIN_PREFIX}
+    --pytorch_deeplab_model ${TECA_DATA_ROOT}/cascade_deeplab_IVT.pt
```

Comment:
option names don't change & there is no need to pass the option name as an argument. for example here you are passing `--pytorch_deeplab_model ${TECA_DATA_ROOT}/cascade_deeplab_IVT.pt` instead you can just pass `${TECA_DATA_ROOT}/cascade_deeplab_IVT.pt` (please change this here and elsewhere where you have done that).

Reply:
Don't you think it's cleaner that way? where any developer can tell what arguments were passed to the app in the cmake file without having to look into the bash file and try to connect the options and values together

Reply:
no, I would not have asked you to change it if I thought that. it makes everything longer than it needs to be. the cmake script and you shell scripts will be significantly shorter and easier to read. it's something that I would be tempted to let slide but since you have more work to do, we might as well clean it up.

Reply:
btw, I have similar thoughts about ${app_exec} in your shell script. Using a variable here adds an unnecessary line of code, and makes the necessary line of code (the one that runs the app) less clear.

Reply:
makes sense

Reply:
see 374ab79 , addresses these issues for the Bayesian AR detector test